### PR TITLE
fix(ingestion): Redshift usage refactoring - simplify, annotate, fix bugs

### PR DESCRIPTION
**Pull request by rslanka**
This PR refactors the redshift_usage source to
 - make the code simpler and more readable
 - enhance annotations
 - fix subtle bugs, replacing problematic code with idiomatic python.
 
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.

### CODE REVIEW

**Review by aditya-radhakrishnan — APPROVED** [review]
LGTM!

File: metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py [hunk 1]
```diff
@@ -182,18 +173,17 @@ def _make_usage_query(self, query: str) -> str:
...
-               sui.usename,
-               ss.tbl,
-               sq.querytxt,
-               sti.database,
-               sti.schema,
-               sti.table,
-               sq.starttime,
-               sq.endtime,
-               sq.aborted
+        SELECT DISTINCT ss.userid as userid,
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py:176]: We could replace the whole logic of getting insert and delete types separately with one union query like this:
```
SELECT DISTINCT ss.userid as userid,
       ss.query as query,
       ss.rows as rows,
       sui.usename as username,
       ss.tbl as tbl,
       sq.querytxt as querytxt,
       sti.database as database,
       sti.schema as schema,
       sti.table as table,
       sq.starttime as starttime,
       sq.endtime as endtime,
       'insert' as type
FROM stl_insert ss
JOIN svv_table_info sti ON ss.tbl = sti.table_id
JOIN stl_query sq ON ss.query = sq.query
JOIN svl_user_info sui ON sq.userid = sui.usesysid
AND ss.rows > 0
AND sq.aborted = 0
union
SELECT DISTINCT ss.userid as userid,
       ss.query as query,
       ss.rows as rows,
       sui.usename as username,
       ss.tbl as tbl,
       sq.querytxt as querytxt,
       sti.database as database,
       sti.schema as schema,
       sti.table as table,
       sq.starttime as starttime,
       sq.endtime as endtime,
       'delete' as type
FROM stl_delete ss
JOIN svv_table_info sti ON ss.tbl = sti.table_id
JOIN stl_query sq ON ss.query = sq.query
JOIN svl_user_info sui ON sq.userid = sui.usesysid
AND ss.rows > 0
AND sq.aborted = 0
ORDER by endtime desc
```

**Review by treff7es — APPROVED** [review]
lgtm

File: metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py (hunk 1 above)
Comment (rslanka) [reply metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py:176]: This is a great idea! Will do it in  a separate PR.

**Pull request merged**